packages/notebook/src/widget.ts

```diff
@@ -1033,6 +1036,11 @@ export namespace StaticNotebook {
      * The application language translator.
      */
     translator?: ITranslator;
+
+    /**
+     * The kernel history retrieval object
+     */
+    history?: INotebookHistory;
```

Comment:
Maybe worth being more specific as `inputHistory`? The other kinds that quickly come to mind are "editHistory" (undo stack), "checkpointHistory". I think this should be reflected in the name of attributes and interface.

Reply:
Maybe `kernelHistory` or `executedInputHistory`?

Reply:
`kernelHistory` sounds good to me